aten/src/ATen/core/jit_type.h

```diff
@@ -676,6 +619,12 @@ struct CAFFE2_API ProfiledTensorType : public TensorType {
   VaryingShape sizes_;
   VaryingStrides strides_;
   c10::optional<bool> requires_grad_;
+  // we exploit the fact certain tensors must be zero in the autograd to 
+  // optimize gradient computation. If true, this means that this tensor
+  // must only contain zeros. Normally this will be nullopt, meaning
+  // the tensor may or may not contain only zeros. If false, 
+  // this means the tensor must have some non-zero elements.
+  c10::optional<bool> autograd_zero_;
```

Comment:
I wish we had this comment before, took me a while to figure out how all these pieces fit together 😢